class.jetpack-gutenberg.php

```diff
@@ -922,6 +922,30 @@ public static function validate_block_embed_url( $url, $allowed = array(), $is_r
 		return false;
 	}
 
+	/**
+	 * Determines whether a preview of a block should be displayed on the
+	 * frontend of a site.
+	 *
+	 * @since 8.4.0
+	 *
+	 * @param string $slug Slug for the block.
+	 *
+	 * @return bool
+	 */
+	public static function should_show_frontend_preview_for_block( $slug ) {
+		$availability = self::get_availability();
+
+		return (
+			isset( $availability[ $slug ]['details']['required_plan'] ) &&
+			current_user_can( 'manage_options' )
+			/** This filter is documented in class.jetpack-gutenberg.php */
+			&& apply_filters( 'jetpack_block_editor_enable_upgrade_nudge', false )
+			/** This filter is documented in _inc/lib/admin-pages/class.jetpack-react-page.php */
+			&& apply_filters( 'jetpack_show_promotions', true )
+			&& ! is_feed()
```

Comment:
The last two conditions here were part of the original check for rendering a frontend upgrade nudge. I am looking for more context to see if they are still needed, particularly the `jetpack_show_promotions` filter.

Reply:
I'm not seeing any usage of the `jetpack_show_promotions` filter across the Jetpack or WordPress.com codebase. I suspect this is a global way to turn off all promotional notifications across Jetpack. It's possible that this is being used by hosts somewhere though, so best to leave it in.

Reply:
`jetpack_block_editor_enable_upgrade_nudge` is the current way we enable the display of upgrade nudges on WordPress.com. Do we still need this with this work?